src/rgw/rgw_sal_motr.cc

```diff
@@ -2484,10 +2484,12 @@ int MotrAtomicWriter::complete(size_t accounted_size, const std::string& etag,
   if (rc == 0)
     store->get_obj_meta_cache()->put(dpp, obj.get_key().to_str(), bl);
 
-  if (old_obj.get_bucket()->get_info().versioning_status() != BUCKET_VERSIONED) {
-    // Delete old object data if exists.
-    old_obj.delete_mobj(dpp);
-  }
+  if (rc == 0 && is_versioned)
```

Comment:
`is_versioned` does not necessarily say whether bucket versioning is enabled or not. I think the previous check for bucket versioning was more appropriate. We can have additional check in the previous if statement itself.

Reply:
Ok, two questions then:
1) If we need to check bucket - why not to use `bucket->versioning_enabled()`?
2) Should we set `is_versioned` flag above using the same method?

Reply:
See the updated version of the patch.